src/figuresClasses.ts

```diff
@@ -2,10 +1,77 @@
+type Shapes = 'triangle' | 'rectangle' | 'circle';
+type Colors = 'red' | 'blue' | 'green';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shapes = 'triangle';
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Values must be positive numbers, greater than zero');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('The provided sides do not form a valid triangle');
```

Comment:
The error message 'The provided sides do not form a valid triangle' is correct, but ensure it aligns with any specific error messaging requirements in the task description or checklist.